Userland/Libraries/LibHTTP/HttpRequest.cpp

```diff
@@ -153,16 +189,12 @@ Optional<HttpRequest> HttpRequest::from_raw_request(ReadonlyBytes raw_request)
         }
     }
 
-    HttpRequest request;
-    if (method == "GET")
-        request.m_method = Method::GET;
-    else if (method == "HEAD")
-        request.m_method = Method::HEAD;
-    else if (method == "POST")
-        request.m_method = Method::POST;
-    else
+    auto http_method = parse_request_method(method);
+    if (!http_method.has_value()) {
         return {};
-
+    }
```

Comment:
Right, but it prevents certain kind of mistakes, see https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces for a longer discussion. 

I couldn't find this particular style preference documented either in the style code nor in the clang-format options. Wouldn't it be better to enforce this kind of styling via tools instead of discussing it in a pull request?

Reply:
Unfortunately the tooling needed to enforce this kind of style needs to be context-aware (i.e. clang-tidy instead of clang-format, which is what we currently yes), but in our experience they are a bit too false-positive prone to use in CI. I agree that it should be documented, but it is still the current style choice.